### DISCUSSION TITLE
Reusable items

### DISCUSSION THREAD

**Discussion by ldimitrijevic** [OrchardCMS/OrchardCore#13315] [Q&A]
Hello,
i've been working on a site for one of my client and there are blocks which are repeated in mutliple pages within the content. It's a testomony part and the client wants that part on the home page and on an inner page. It is not aside so i don't think i can use zones to manage this.
The options are simple:
- either i duplicate this section which is not nice to administrate
- Or... i don't know.
How is it possible to have content items that can be inserted within multiple pages. Is it possible (i don't think so) to have a content part with values already set?

**Comment by jtkech** [comment]
One solution is to have a dedicated type having that part and create content item(s) of that type, then in other content types add a `ContentPickerField` (for this dedicated type), so that a given content item could point to the item you want.

Then, the related default `ContentPickerField` view will render the item(s) in the summary display type, only rendering link(s) using the display text(s), so you may need to override this `ContentPickerField` view for this display mode and/or only for some content types (this by using the right view alternate names), so that you can render what you want.

---

There are other things that could be done but not so easy, you could have an index provider to retrieve only the data you need in some Dtos objects from an index table in place of loading full items, you can also use a cache deriving from our `IVolatileDocumentManager`, so that you load these Dtos once and cache them in memory, but here would need a content handler to invalidate the cache when you update an item of the dedicated type, so that's another story ;)

**Comment by ldimitrijevic** [comment]
Thanks for the reply. I did not know about Content Picker. Does it picks all the fields (kind of a copy) or is it a reference? The documentation is "empty" for it.

**Reply by jtkech** [reply]
It is a reference holding the ContentItemId(s) of the referenced item(s), then see in `ContentPickerField.cshtml` how the related item(s) are retrieved.

Just tried with a CPField named CP1 used in the Article ContentType, here the alternates you can use in your theme to override the default view. From memory, to get a view name from an alternate you have to replace a single underscore with a dot `.` and double underscores `__` with a dash `-`, also the display type as `Detail` should be at the end. Also, good to use a more specific alternate name, for example containing a content type or the name of your CPField, so that you don't override all CPFields.

![CPField Alternates](https://user-images.githubusercontent.com/8586360/221670628-2f036a07-7d5e-4962-b285-c574abf52af8.png)

For example `Article_Detail__ContentPickerField` => `Article-ContentPickerField.Detail.cshtml`.
